packages/components/src/utils/getTextAsHtml.ts

```diff
@@ -20,31 +22,79 @@ export const getTextAsHtml = (content?: (HardBreakProps | TextProps)[]) => {
     return ""
   }
 
-  return content
-    .map((node) => {
-      if (node.type === "hardBreak") {
-        return "<br />"
-      }
+  const output: string[] = []
+  let existingLinkMark: Marks | undefined = undefined
+
+  // At every step, we will close off all marks except for links
+  // First encounter with a link, always open it first before other marks
+  // Close all other marks first before closing the link mark
+  content.forEach((node) => {
+    if (node.type === "hardBreak") {
+      output.push("<br />")
+      return
+    }
+
+    const newLinkMark = node.marks?.find((mark) => mark.type === "link")
+    const isLinkMarkNew =
+      !!newLinkMark && !isEqual(existingLinkMark, newLinkMark)
+
+    // Close off the existing link mark if it is different
+    if (isLinkMarkNew && !!existingLinkMark) {
+      output.push(`</${MARK_DOM_MAPPING.link}>`)
+      existingLinkMark = undefined
+    }
 
-      if (!node.marks) {
-        return node.text
-      }
+    // If there are no marks, just push the text
+    if (!node.marks) {
+      output.push(node.text)
+      return
+    }
+
+    if (isLinkMarkNew) {
+      existingLinkMark = newLinkMark
+
+      // Sort such that the link mark is the first item
+      node.marks.sort((a, b) => {
+        if (a.type === "link") {
+          return -1
+        }
+
+        return 1
+      })
 
-      let output = node.text
       node.marks.forEach((mark) => {
         if (mark.type === "link") {
-          output = `<${MARK_DOM_MAPPING[mark.type]} href="${
-            mark.attrs.href
-          }">${output}</${MARK_DOM_MAPPING[mark.type]}>`
-          return
+          output.push(
+            `<${MARK_DOM_MAPPING[mark.type]} target="${mark.attrs.target ?? "_self"}" href="${mark.attrs.href}">`,
+          )
+        } else {
+          output.push(`<${MARK_DOM_MAPPING[mark.type]}>`)
         }
+      })
+    } else {
+      // Continue with the rest of the marks
+      node.marks
+        .filter((mark) => mark.type !== "link")
+        .forEach((mark) => {
+          output.push(`<${MARK_DOM_MAPPING[mark.type]}>`)
+        })
+    }
```

Comment:
this takes place inside a `forEach`? should we just be doing `output.push` here since the `mark.type !== link` is basically guaranteed

my confusion here stems from the fact that eg: if we have a mark array like `[bold, italics, underline]`, this would first go through the initial array, then push each non link item **at each iteration** such that the end output would be `[bold, italics, underline] * 3`

Reply:
Clarified offline, this is because our approach is to always open and close all marks other than link (and close in the correct order which was fixed in 4127bada6311bf562b2500609445d16bbce877dc). Hence, whenever we iterate and go into a new text node in the next iteration, we are always starting "fresh" with the exception of the link node which we are tracking separately.